tests/DCPS/WaitForAckRace/publisher.cpp

```diff
@@ -0,0 +1,256 @@
+#endif
+
+#include <tests/Utils/ExceptionStreams.h>
+#include <tests/Utils/StatusMatching.h>
+
+#include <ace/streams.h>
+#include <ace/Get_Opt.h>
+#include <ace/OS_NS_unistd.h>
+
+#include <memory>
+#include <sstream>
+
+using namespace Messenger;
+using namespace std;
+
+int wait_match(const DDS::DataWriter_var& writer, unsigned int current_readers, unsigned int total_readers)
+{
+  int ret = -1;
+  DDS::StatusCondition_var condition = writer->get_statuscondition();
+  condition->set_enabled_statuses(DDS::PUBLICATION_MATCHED_STATUS);
+  DDS::WaitSet_var ws(new DDS::WaitSet);
+  DDS::ReturnCode_t a = ws->attach_condition(condition);
+  if (a != DDS::RETCODE_OK) {
+    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: %N:%l: wait_match() - attach_condition returned %d\n"), a));
+    return ret;
+  }
+
+  const CORBA::Long n_current_readers = static_cast<CORBA::Long>(current_readers);
+  const CORBA::Long n_total_readers = static_cast<CORBA::Long>(total_readers);
+  const DDS::Duration_t forever = { DDS::DURATION_INFINITE_SEC, DDS::DURATION_INFINITE_NSEC };
+  DDS::PublicationMatchedStatus ms = { 0, 0, 0, 0, 0 };
+  DDS::ConditionSeq conditions;
+  while (ret != 0) {
+    if (writer->get_publication_matched_status(ms) == DDS::RETCODE_OK) {
+      if (ms.current_count == n_current_readers && ms.total_count == n_total_readers) {
+        ret = 0;
+      } else { // wait for a change
+        DDS::ReturnCode_t w = ws->wait(conditions, forever);
+        if ((w != DDS::RETCODE_OK) && (w != DDS::RETCODE_TIMEOUT)) {
+          ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: %N:%l: wait_match() - wait returned %d\n"), w));
+          break;
+        }
+      }
+    } else {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: %N:%l: wait_match() - get_publication_matched_status failed!\n")));
+      break;
+    }
+  }
+
+  ws->detach_condition(condition);
+  return ret;
+}
+
+int ACE_TMAIN(int argc, ACE_TCHAR *argv[])
+{
+  sigset_t mask, prev;
+  ACE_OS::sigemptyset(&mask);
+  ACE_OS::sigaddset(&mask, SIGPIPE);
+  ACE_OS::sigprocmask(SIG_BLOCK, &mask, &prev);
+
+  int total_readers = 1;
+
+  bool large_samples = false;
+  bool sub_delayed_exit = false;
+  for (int i = 0; i < argc; ++i) {
+    if (std::string(ACE_TEXT_ALWAYS_CHAR(argv[i])) == "-l") {
+      large_samples = true;
+    }
+    if (std::string(ACE_TEXT_ALWAYS_CHAR(argv[i])) == "-d") {
+      sub_delayed_exit = true;
+    }
+    if (std::string(ACE_TEXT_ALWAYS_CHAR(argv[i])) == "-r") {
+      ++i;
+      if (i < argc) {
+        total_readers = ACE_OS::atoi(argv[i]);
+      }
+    }
+  }
+
+  try
+  {
+    DDS::DomainParticipantFactory_var dpf =
+      TheParticipantFactoryWithArgs(argc, argv);
+
+    DDS::DomainParticipant_var participant =
+      dpf->create_participant(31,
+                              PARTICIPANT_QOS_DEFAULT,
+                              DDS::DomainParticipantListener::_nil(),
+                              ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
```

Comment:
```suggestion
                              0,
                              0);
```

It would be good find an example to work off of that doesn't have so much unnecessary cruft in it.  All the `.in()` and `is_nil`, `_nil()`, etc...
Otherwise the old patterns of usage just keep getting propagated